packages/react/src/analytics/integrations/Guardian/providers/__tests__/Riskified.test.ts

```diff
@@ -0,0 +1,125 @@
+import { EnvironmentTypes, integrations } from '@farfetch/blackout-analytics';
+import { Riskified } from '..';
+
+describe('Riskified', () => {
+  const defaultKey = 'farfetch.com';
+  const baseUrl = (key?: string) =>
+    `https://beacon.riskified.com?shop=${key || defaultKey}&sid=`;
+  const localId = '3c11a6f9-6d5e-45c4-b1b2-d26736964605';
+
+  const loadData = {
+    user: {
+      localId,
+    },
+  };
+  describe('Riskified Integration', () => {
+    let instance: Riskified;
+    const options = { environment: EnvironmentTypes.PROD };
+
+    beforeEach(() => {
+      instance = Riskified.createInstance(
+        options,
+        // @ts-expect-error simple integration data
+        loadData,
+        undefined,
+      ) as Riskified;
+    });
+
+    it('Should extend the abstract class `Integration`', () => {
+      expect(Riskified.prototype).toBeInstanceOf(integrations.Integration);
+    });
+
+    it('Should be ready to load by default', () => {
+      expect(Riskified.shouldLoad()).toEqual(true);
+    });
+
+    it('Should have a base URL defined', () => {
+      expect(instance.baseUrl).toEqual(baseUrl());
+    });
+
+    it('Should set the script and append it to the DOM', () => {
+      const script: HTMLScriptElement | null = document.querySelector(
+        '[data-test="riskified"]',
+      );
+
+      expect(script).toBeDefined();
+      expect(script?.async).toBe(true);
+      expect(script?.src.endsWith(localId)).toBeTruthy();
+    });
+
+    it('Should set to true when onLoad occurs', () => {
+      expect(instance.isReady).toBe(false);
+
+      instance.onLoad();
+
+      expect(instance.isReady).toBe(true);
+    });
+
+    it('Should call `RISKX.go` if the integration is ready', () => {
+      const data = {
+        type: 'page',
+        context: {
+          web: {
+            window: {
+              location: {
+                href: 'http://localhost',
+              },
+            },
+          },
+        },
+        user: {
+          localId,
+        },
+      };
+      const spy = jest.fn();
+
+      window.RISKX = {
+        go: spy,
+      };
+
+      instance.isReady = true;
+      // @ts-expect-error
+      instance.track(data);
+
+      expect(spy).toBeCalledWith(data.context.web.window.location.href);
+    });
+
+    it('Should not call `RISKX.go` if the integration isnt ready', () => {
+      const data = {
+        type: 'page',
+        properties: {
+          location: {
+            href: 'http://localhost',
+          },
+        },
+        user: {
+          localId,
+        },
+      };
+      const spy = jest.fn();
+
+      window.RISKX = {
+        go: spy,
+      };
+
+      instance.isReady = false;
+      // @ts-expect-error
+      instance.track(data);
+
+      expect(spy).not.toBeCalled();
+    });
+  });
+  describe('Riskified Integration without defaults', () => {
+    it('Should use not default siteId', () => {
```

Comment:
```suggestion
    it('Should not use the default siteId if one is passed in options', () => {
```

Reply:
Now without defaults to Shop property, this case will be deleted